### PR TITLE
Add size control system

### PR DESCRIPTION
**Pull request by knelli2**
## Proposed changes

Also replace some interpolated quantities in the CharSpeed measurement. The current ones are no longer necessary. And some other small miscellaneous changes.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/DataStructures/LinkedMessageQueue.hpp [hunk 1]
```diff
@@ -34,6 +34,7 @@ class LinkedMessageQueue<Id, tmpl::list<QueueTags...>> {
 
  public:
   using IdType = Id;
+  using QueueTagsList = tmpl::list<QueueTags...>;
```
Comment (wthrowe) [inline src/DataStructures/LinkedMessageQueue.hpp]: Snake case for a metavariable.

File: src/ControlSystem/Systems/Size.hpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+ * - Currently this control system can only be used with the \link
+ *   control_system::ControlErrors::Size Size \endlink control error
+ */
+template <::domain::ObjectLabel Horizon, size_t DerivOrder>
+struct Size : tt::ConformsTo<protocols::ControlSystem> {
+  static constexpr size_t deriv_order = DerivOrder;
+
+  static std::string name() { return "Size"s + ::domain::name(Horizon); }
+
+  static std::optional<std::string> component_name(
```
Comment (wthrowe) [inline src/ControlSystem/Systems/Size.hpp:68]: Why is this an optional?  I know the protocol says it is, but I can't find any documentation on what an empty return would mean.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: src/ControlSystem/IsSize.hpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+#pragma once
+
+#include "ControlSystem/Systems/Size.hpp"
+#include "Domain/Structure/ObjectLabel.hpp"
+
+namespace control_system::size {
+// tt::is_a doesn't work because of domain::ObjectLabel and size_t
+template <template <domain::ObjectLabel, size_t> class U, typename T>
+struct is_size : std::false_type {};
```
Comment (wthrowe) [inline src/ControlSystem/IsSize.hpp:12]: Do you need the first template parameter?  Looks like it's always the same, and it would make `is_size` and `is_size_v` consistent.

**Review by wthrowe — CHANGES_REQUESTED** [review]
Missed a commit in my previous review.

File: src/ControlSystem/Systems/Size.hpp (hunk 2 above)
Comment (knelli2) [reply src/ControlSystem/Systems/Size.hpp:68]: This is because of shape control. I added a new commit that adds some docs to the protocol.

**Comment by knelli2** [comment]
Squashed in the changes and added a new commit (the most recent one)

**Review by wthrowe — APPROVED** [review]

File: src/ControlSystem/Systems/Size.hpp [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+      Strahlkorper<Frame::Distorted> distorted_excision_surface{};
+      strahlkorper_in_different_frame_aligned(
+          make_not_null(&distorted_excision_surface), grid_excision_surface,
+          Parallel::get<domain::Tags::Domain<3>>(cache),
+          Parallel::get<domain::Tags::FunctionsOfTime>(cache),
+          measurement_id.id);
+
+      // excision strahlkorper
+      Parallel::simple_action<::Actions::UpdateMessageQueue<
```
Comment (nilsvu) [inline src/ControlSystem/Systems/Size.hpp]: Why are there so many messages sent? Can't the data be sent in one message?

File: src/ControlSystem/Systems/Size.hpp [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+ * - This control system requires that there be at least one object in the
+ *   simulation
+ * - This object must be a black hole (have an excision)
+ * - Currently this control system can only be used with the \link
+ *   control_system::measurements::CharSpeed CharSpeed \endlink
+ * measurement
+ * - Currently this control system can only be used with the \link
+ *   control_system::ControlErrors::Size Size \endlink control error
+ */
+template <::domain::ObjectLabel Horizon, size_t DerivOrder>
```
Comment (nilsvu) [inline src/ControlSystem/Systems/Size.hpp:62]: How would I choose the deriv order? Maybe add docs here or to the protocol (can be another PR)

**Review by nilsvu — COMMENTED** [review]
Just questions, don't hold up this PR

File: src/ControlSystem/Systems/Size.hpp (hunk 4 above)
Comment (wthrowe) [reply src/ControlSystem/Systems/Size.hpp]: It didn't occur to me that these were all triggering separate messages.  While that's not as critical for control systems, you're right, they should be merged.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: src/ControlSystem/Systems/Size.hpp (hunk 4 above)
Comment (knelli2) [reply src/ControlSystem/Systems/Size.hpp]: The `UpdateMessageQueue` action currently only takes one queue tag at a time. This can of course be changed, but I'll do that in a separate PR.

Comment (wthrowe) [reply src/ControlSystem/Systems/Size.hpp]: I was thinking just storing a tuple.

File: src/ControlSystem/Systems/Size.hpp (hunk 5 above)
Comment (knelli2) [reply src/ControlSystem/Systems/Size.hpp:62]: By trial and error basically. Or know that deriv order 2 was found to work well in spec for most control systems (except a notable couple). Either way, I'll do this in another PR. I know I need to add an entire explanation on how to control system communication works anyways.

File: src/ControlSystem/Systems/Size.hpp (hunk 4 above)
Comment (nilsvu) [reply src/ControlSystem/Systems/Size.hpp]: @wthrowe are you requesting changes for this PR?

Comment (wthrowe) [reply src/ControlSystem/Systems/Size.hpp]: Instead of having five separate entries in the queue, have one that stores a tagged tuple or a struct or something.

**Comment by knelli2** [comment]
@wthrowe Added new queue tags that hold TaggedTuples of the other tags. Now size only sends one message per submeasurement.

**Comment by wthrowe** [comment]
Looks good.  Squash.

**Review by wthrowe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]
